### DISCUSSION TITLE
Hot-restart does not stop loading when using streams

### DISCUSSION THREAD

**Discussion by CodeDoctorDE** [fzyzcjy/flutter_rust_bridge#1340] [Q&A]
Hello,
I don't know if it is a bug or not but I have a problem.
If I leave the widget that subscribed to my StreamSink, it successfully closed the stream.
But when I click on the hot restart button it doesn't stop loading and the dispose function of the function won't be executed.
How can I fix this?
My code:
* The widget: https://github.com/LinwoodDev/cuboverse/blob/2b7bdde6cb4d1cb547590e8294f68f9cbb9b8d32/app/lib/pages/game/page.dart#L30
* The rust code behind: 
  * Close: https://github.com/LinwoodDev/cuboverse/blob/2b7bdde6cb4d1cb547590e8294f68f9cbb9b8d32/app/native/src/api.rs#L76
  * Create: https://github.com/LinwoodDev/cuboverse/blob/2b7bdde6cb4d1cb547590e8294f68f9cbb9b8d32/app/native/src/api.rs#L72

**Comment by fzyzcjy** [comment]
Hi, it seems that your sample code is quite complex. Could you please provide a minimal reproducible sample?

I did not checked into details, but maybe firstly do abundant logging to see where it is stuck, eg a deadlock?

Note that hot restart only restarts dart vm, while the rust code's state is *unchanged*
